<a href="https://colab.research.google.com/github/davidmiheev/QuantQuestions/blob/main/Prob_Questions_%5BPart_1%5D.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Probability questions from Quant Questions: Part 1
Here are collected problems where you are asked to find some probability or expectation value and my solutions for these problems

## Increasing Dice Order I

> You roll a fair 6-sided die twice.
Calculate the probability that the value of the first roll is strictly less than the value of the second roll.

### Solution

Use Full Probabily Formula:

Let $X_1$ - value of the first roll and $X_2$ - value of the second roll

Then we need to find probability that $X_2 > X_1$:
$$P(X_2>X_1) = \sum_{k=1}^6 P(X_2>X_1|X_1=k)P(X_1=k)$$

We know that $P(X_1=k)=1/6$ for any $k$
and $P(X_2>X_1|X_1=k)=(6-k)/6$

Now, we can substitute all values to the formula and get:
$$P(X_2>X_1) = 1/6\sum_{k=1}^6 (6-k)/6=5/12$$

So, the answer is $5/12$



In [7]:
from fractions import Fraction

print('The answer is', Fraction('1/6')*sum(Fraction(f'{6-k}/6') for k in range(1, 7)))

The answer is 5/12


## Probability of Unfair Coin I

> You have a pile of 100 coins.
1 of the coins is an unfair coin and has heads on both sides. The remaining
99 coins are fair coins. You randomly select a coin from the pile and flip it
10 times. The coin lands heads all
10 times. Calculate the probability that the coin you selected is the unfair coin.

### Solution

Use Bayes Formula
$$P(A|B)=\frac{P(B|A)\cdot P(A)}{P(B)},$$

where A is the event "selected unfair coin", B is the event "The coin lands heads all 10 times"

Then $P(B|A) = 1, P(A)=1/100, P(B)=1/2^{10}\cdot 99/100+1/100$
So,
$$P(A|B)=1\cdot 1/100 \cdot \frac{1024\cdot 100}{1123}= \frac{1024}{1123}$$

the answer is 1024/1123

In [8]:
print('The answer is', Fraction('1024/1123'))

The answer is 1024/1123


## First Ace

> On average, how many cards in a normal deck of 52 playing cards do you need to flip over to observe your first ace?

### Solution 1: Python script

Let $X$ is the number of cards you need to flip over to observe first ace
We need to find expectation of $X$

Use definition of expected value:
$$E[X]=\sum_k kP(X=k)$$,
where $P(X=k)= \frac{4}{52-(k-1)}\prod_{j=1}^{k-1}\frac{52-4-(j-1)}{52-(j-1)}$

See the script, that computes this sum below

In [9]:
expected_value = Fraction('0')
prob = Fraction('1')
for k in range(1, 53):
  expected_value += Fraction(f'{4*k}/{53-k}')*prob
  prob *= Fraction(f'{49-k}/{53-k}')

print('The answer is', expected_value, 'or', float(expected_value))

The answer is 53/5 or 10.6


### Solution 2: Linearity of Expectation

Let $L_i, i=1...5$ is a length of interval $(A_{i-1}, A_i)$, where $A_i$ is a position of i-th ace in the deck ($A_0$ is the beginning of the deck and $A_5$ is the end of the deck. Because of symmetry $E[L_1] = E[L_2]=...= E[L_5]$ and we know that $$48=E[L]=E[\sum L_i]=\sum E[L_i] = 5E[L_1],$$
because of linearity of expectation.

So, $E[L_1] = 48/5$, then $E[X]=48/5+1=53/5=10.6$


In [11]:
expected_value = Fraction('53/5')
print('The answer is', expected_value, 'or', float(expected_value))

The answer is 53/5 or 10.6
